arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/raise/Raise.kt

```diff
@@ -654,13 +655,17 @@ public inline fun <Error, B : Any> Raise<Error>.ensureNotNull(value: B?, raise:
  * <!--- TEST lines.isEmpty() -->
  */
 @RaiseDSL
+@Suppress("WRONG_INVOCATION_KIND")
 public inline fun <Error, OtherError, A> Raise<Error>.withError(
```

Comment:
For interest, this version does not need suppression (it uses `merge`, but it's actually the `attempt` function I proposed before):
```
@RaiseDSL
public inline fun <Error, OtherError, A> Raise<Error>.withError(
  transform: (OtherError) -> Error,
  @BuilderInference block: Raise<OtherError>.() -> A
): A {
  contract {
    callsInPlace(transform, AT_MOST_ONCE)
    callsInPlace(block, EXACTLY_ONCE)
  }
  val error = merge { return block(this) }
  raise(transform(error))
}
```
The compiler understands here that the only way the function returns is if `block` is executed.